src/mpz.rs

```diff
@@ -103,9 +105,21 @@ pub struct Mpz {
 
 unsafe impl Send for Mpz { }
 unsafe impl Sync for Mpz { }
-
+use std::sync::atomic;
 impl Drop for Mpz {
-    fn drop(&mut self) { unsafe { __gmpz_clear(&mut self.mpz) } }
+    fn drop(&mut self) {
+        unsafe {
+            let size_limbs = __gmpz_size(&mut self.mpz);
+            let dst = self.mpz._mp_d as *mut c_int;
```

Comment:
Good question! 
I will add a comment here as to why this.
the mpz_limbs_write is not doing compiler fencing(optimizations by compilers might get rid of the code or delay it) which is what we want to do a best effort securely wipe off the underlying limb array.
Also:
1. In gmp source the mp_limb_t is defined as unsigned long so c_int will never be more than that on any platform.
2. Memory for the array is guaranteed to be allocated by gmp in limb sizes. So we can be sure that we are not writing what we should not.

Also I think we should be having something in the gmp library itself to do this kind of a thing. I will try to do a PR for them and if that is accepted, We can remove it here.

Reply:
Actually I looked the source of mpz_limbs_write just now and looks like it makes a new alloc and does not return the internal pointer. That is probably not what we want.
```
#include "gmp-impl.h"

mp_ptr
mpz_limbs_write (mpz_ptr x, mp_size_t n)
{
  ASSERT (n > 0);
  return MPZ_NEWALLOC (x, n);
}
```

Reply:
> looks like it makes a new alloc

As I understand, it will make new alloc only if n > size(x), according to macro definition:

```c
#define MPZ_REALLOC(x, n) \
  (ABSIZ(x) >= (n) ? PTR(x) : (_mpz_realloc ((x),(n)), PTR (x)))

#define MPZ_NEWALLOC MPZ_REALLOC
```

> mpz_limbs_write is not doing compiler fencing

Maybe I missing something, but I thought that `mpz_limbs_write` just returns a pointer to underlying limb array, which should be the same thing as casting pointer (you can still use `std::ptr::write_volatile` and `atomic::{fence, compiler_fence}`), but this function is provided by API, so any further changes in internal representation of GMP number will not affect this code. 

However, I do not believe that internal representation of GMP number might be changed, I think it's okay to leave pointer casting here.